server/src/helper/auth.ts

```diff
@@ -0,0 +1,27 @@
+import { promisify } from 'util';
+import { RedisClient } from 'redis';
+import Container from 'typedi';
+import config from '@/config';
+
+export const storeRefreshToken = (refresh: string, idx: number) => {
+  const redisClient = Container.get<RedisClient>('redisClient');
+  const setexAsync = promisify(redisClient.setex).bind(redisClient);
+  return setexAsync(
+    refresh,
+    60 * 60 * config.jwt.expire.refresh,
+    idx.toString(),
+  );
+};
+
+export const deleteRefreshToken = (refresh: string) => {
+  const redisClient = Container.get<RedisClient>('redisClient');
+  const delAsync = promisify(redisClient.del).bind(redisClient);
+  return delAsync(refresh);
+};
+
+export const verifyRefreshToken = async (token: string, idx: number) => {
+  const redisClient = Container.get<RedisClient>('redisClient');
+  const getAsync = promisify(redisClient.get).bind(redisClient);
+  const res = await getAsync(token);
+  return res ? parseInt(res, 10) === idx : false;
```

Comment:
예상치 못한 오류를 대비해 Number를 사용하는 것이 좋아 보입니다.
https://github.com/denysdovhan/wtfjs#parseint-is-a-bad-guy

```suggestion
  return res ? Number(res, 10) === idx : false;
```

Reply:
그렇네요, 이 부분은 Number를 사용해서 정확하게 체크하는 게 좋아 보이네요! 좋은 지적인 것 같습니다

Reply:
이 부분에 대해서 좀 더 찾아봤는 데 해당 부분은 parseInt가 더 맞는 것 같습니다. 남겨주신 글에서 문제가 생기는 부분을 테스트해봤는 데 10진수로 했을 때는 정상적으로 NaN을 리턴하고 있었습니다.

```ts
parseInt("f*ck", 10); -> NaN
parseInt(null, 10); -> NaN
```

오히려 Number를 사용하게 되면 발생할 수 있는 문제가 아래처럼 null, true, 등에 대해서도 0 or 1 같은 정수값을 반환하고 있다는 것입니다. parseInt를 사용하는 쪽이 더 엄격하게 변환한다라고 생각해서 Number 보다는 parseInt를 위에서는 사용하는 게 좋다고 생각합니다.

```ts
parseInt();         // NaN
parseInt(null);     // NaN
parseInt(true);     // NaN
parseInt("");       // NaN

Number();           // 0
Number(null);       // 0
Number(true);       // 1
Number("");         // 0
```

참고 : [Number() vs parseInt() - this vs that](https://thisthat.dev/number-constructor-vs-parse-int/)